### ISSUE TITLE
`AWS::StepFunctions::StateMachine`: cfn-lint doesn't recognize the new `Assign:` (variables) feature

### ISSUE THREAD

**Issue by r-heimann** [aws-cloudformation/cfn-lint#3858]
### CloudFormation Lint Version

cfn-lint 1.20.2

### What operating system are you using?

Windows 11

### Describe the bug

Step Functions now support variables for JSONPatch and JSONata. I tried to use Variables with JSONPatch, but cfn-lint tells me

```
[cfn-lint] E3601: Additional properties are not allowed ('Assign' was unexpected)
```

I was able to manually deploy the CloudFormation template that includes `Assign`.

### Expected behavior

cfn-lint recognized `Assign`.

### Reproduction template

```yaml
AWSTemplateFormatVersion: 2010-09-09
Description: Test

Resources:
  StepFunctionRole:
    Type: AWS::IAM::Role
    Properties:
      AssumeRolePolicyDocument:
        Statement:
          - Effect: Allow
            Principal:
              Service: states.amazonaws.com
            Action: sts:AssumeRole
      Policies:
        - PolicyName: "InlinePolicy"
          PolicyDocument:
            Version: 2012-10-17
            Statement:
              - Sid: "RDSAccess"
                Effect: Allow
                Action:
                  - rds:CreateDBSnapshot
                  - rds:AddTagsToResource
                  - rds:DescribeDBSnapshots
                Resource:
                  - !GetAtt RDS.DBInstanceArn
                  - !Sub "arn:aws:rds:${AWS::Region}:${AWS::AccountId}:snapshot:*"

  StateMachine:
    Type: AWS::StepFunctions::StateMachine
    Properties:
      RoleArn: !GetAtt StepFunctionRole.Arn
      Definition:
        StartAt: Create DB Snapshot
        States:
          Create DB Snapshot:
            Type: Task
            Resource: arn:aws:states:::aws-sdk:rds:createDBSnapshot
            Parameters:
              DbInstanceIdentifier: !GetAtt RDS.DBInstanceIdentifier
              DbSnapshotIdentifier: TestBackup
            Assign: # <---- This here
              DbSnapshotIdentifierVariable.$: $.DbSnapshot.DbSnapshotIdentifier
            End: true
```
